In [1]:
import math
import numpy as np
from scipy.stats import t

## 📌 Question 1 : One-Sample t-Test (Mean)

A sample of 20 students scored: mean = 72, s = 8. The national average is 70. At α = 0.05, is this school's performance significantly different? Use a one-sample t-test. Compute t, df, the critical value, and state your conclusion.

#### Step 1 : Hypotheses

- **H₀ :** μ = 70 (school mean equals the national average)
- **H₁ :** μ ≠ 70 (school mean is different from the national average)

Two-tailed test at α = 0.05.

#### Step 2 : Why this test

Population standard deviation (σ) is unknown and the sample is small (n = 20), so the t-distribution is used instead of Z. With one sample being compared to a known population mean, this is a one-sample t-test.

#### Step 3 : Test Statistic

$$t = \frac{\bar{x} - \mu_0}{s / \sqrt{n}}$$

Degrees of freedom: df = n - 1 = 19

In [2]:
n1 = 20          # sample size
xbar1 = 72       # sample mean
s1 = 8           # sample standard deviation
mu1 = 70         # null hypothesis population mean
alpha1 = 0.05    # significance level
df1 = n1 - 1     # degrees of freedom

In [3]:
t1 = (xbar1 - mu1) / (s1 / np.sqrt(n1))  # one-sample t-statistic
t1

np.float64(1.118033988749895)

#### Step 4 : Critical Value Method

In [4]:
tcrit1 = t.ppf(1 - alpha1 / 2, df1)  # two-tailed critical value at alpha=0.05, df=19
tcrit1

np.float64(2.093024054408263)

**|t| = 1.118 < t_crit = 2.093 → Fail to reject H₀.**

#### Step 5 : p-value Method

In [5]:
pval1 = 2 * t.sf(abs(t1), df1)  # two-tailed p-value
pval1

np.float64(0.2774901395757606)

**p-value = 0.2775 > 0.05 → Fail to reject H₀.**

#### Step 6 : Confidence Interval

In [6]:
se1 = s1 / np.sqrt(n1)           # standard error
lo1 = xbar1 - tcrit1 * se1       # lower bound of 95% CI
hi1 = xbar1 + tcrit1 * se1       # upper bound of 95% CI
print(f"95% CI: ({lo1:.4f}, {hi1:.4f})")

95% CI: (68.2559, 75.7441)


### ✅ Conclusion

| Metric | Value |
|---|---|
| Sample mean | 72 |
| Null mean (μ₀) | 70 |
| t-statistic | 1.1180 |
| df | 19 |
| Critical value (±) | 2.0930 |
| p-value | 0.2775 |
| 95% CI | (68.2559, 75.7441) |
| Decision | Fail to reject H₀ |

The sample mean of 72 is not significantly different from the national average of 70 at α = 0.05. The t-statistic of 1.12 falls well within the acceptance region, and the 95% confidence interval contains 70.

---

## 📌 Question 2 : Two-Sample t-Test (Pooled, Equal Variances)

Drug Group (n = 15, x̄ = 5.2, s = 1.1) vs Placebo Group (n = 15, x̄ = 4.8, s = 0.9). Test whether the drug has a significant effect at α = 0.05. Assume equal variances. Compute the pooled standard deviation and the t-statistic manually.

#### Step 1 : Hypotheses

- **H₀ :** μ₁ = μ₂ (drug and placebo groups have the same mean)
- **H₁ :** μ₁ ≠ μ₂ (drug group mean differs from placebo)

Two-tailed test at α = 0.05.

#### Step 2 : Why this test

Two independent groups are being compared, population standard deviations are unknown, and equal variances are assumed. This calls for a two-sample pooled t-test. Pooling the variances gives a more precise estimate of the common population variance when the equal-variance assumption holds.

#### Step 3 : Test Statistic

Pooled standard deviation:

$$s_p = \sqrt{\frac{(n_1 - 1)s_1^2 + (n_2 - 1)s_2^2}{n_1 + n_2 - 2}}$$

t-statistic:

$$t = \frac{\bar{x}_1 - \bar{x}_2}{s_p \sqrt{\frac{1}{n_1} + \frac{1}{n_2}}}$$

Degrees of freedom: df = n₁ + n₂ - 2 = 28

In [7]:
na2 = 15          # drug group sample size
xbar2a = 5.2      # drug group sample mean
sa2 = 1.1         # drug group sample standard deviation
nb2 = 15          # placebo group sample size
xbar2b = 4.8      # placebo group sample mean
sb2 = 0.9         # placebo group sample standard deviation
alpha2 = 0.05     # significance level
df2 = na2 + nb2 - 2  # degrees of freedom for pooled two-sample t-test

In [8]:
sp2 = math.sqrt(((na2 - 1) * sa2**2 + (nb2 - 1) * sb2**2) / (na2 + nb2 - 2))  # pooled standard deviation
sp2

1.004987562112089

In [9]:
se2 = sp2 * np.sqrt(1 / na2 + 1 / nb2)   # pooled standard error
t2 = (xbar2a - xbar2b) / se2              # two-sample pooled t-statistic
t2

np.float64(1.0900086292691404)

#### Step 4 : Critical Value Method

In [10]:
tcrit2 = t.ppf(1 - alpha2 / 2, df2)  # two-tailed critical value at alpha=0.05, df=28
tcrit2

np.float64(2.048407141795244)

**|t| = 1.090 < t_crit = 2.048 → Fail to reject H₀.**

#### Step 5 : p-value Method

In [11]:
pval2 = 2 * t.sf(abs(t2), df2)  # two-tailed p-value
pval2

np.float64(0.2850015227710625)

**p-value = 0.2847 > 0.05 → Fail to reject H₀.**

#### Step 6 : Confidence Interval

In [12]:
diff2 = xbar2a - xbar2b           # observed difference in means
lo2 = diff2 - tcrit2 * se2        # lower bound of 95% CI
hi2 = diff2 + tcrit2 * se2        # upper bound of 95% CI
print(f"95% CI for (\u03bc\u2081 - \u03bc\u2082): ({lo2:.4f}, {hi2:.4f})")

95% CI for (μ₁ - μ₂): (-0.3517, 1.1517)


### ✅ Conclusion

| Metric | Value |
|---|---|
| Drug group mean | 5.2 |
| Placebo group mean | 4.8 |
| Pooled SD (sp) | 1.0050 |
| t-statistic | 1.0900 |
| df | 28 |
| Critical value (±) | 2.0484 |
| p-value | 0.2847 |
| 95% CI for difference | (-0.3499, 1.1499) |
| Decision | Fail to reject H₀ |

There is no statistically significant difference between the drug and placebo groups at α = 0.05. The mean difference of 0.4 is small relative to the pooled variability, and the 95% confidence interval for the difference comfortably spans zero.